tests/system/providers/amazon/aws/example_bedrock_knowledge_base.py

```diff
@@ -480,6 +483,24 @@ def delete_opensearch_policies(collection_name: str):
     )
     # [END howto_sensor_bedrock_ingest_data]
 
+    # [START howto_operator_bedrock_retrieve_and_generate]
+    retrieve_and_generate = BedrockRaGOperator(
+        task_id="retrieve_and_generate",
+        input="Who was the CEO of Amazon on 2022?",
+        source_type="KNOWLEDGE_BASE",
```

Comment:
Should we have another one that tests the other source type?

Reply:
Sure, I should be able to add that with the s3 data I'm already uploading as part of the test I think... I'll have to experiment a little, but I think I can do it.

Reply:
While adding this I found out that “external sources” was only added to the API 3 weeks ago, meaning it would need botocore min version 1.34.103 current dependency is botocore>=1.33.0 which leaves us with an awkward situation.  We can bump minver on botocore but forcing a 3-week-old version feels aggressive.  

We don't usually do this, but I think in this case maybe wrapping it in a botocore version check and adding a note inthe docs may be the right call?

Reply:
Yeah, I agree with this. Though, I wouldn't wrap in a version check, you can just handle the exception you get back from the boto sdk right?

Reply:
I couuuulllldd.  The error is

```
ParamValidationError: Parameter validation failed:
Unknown parameter in retrieveAndGenerateConfiguration: "externalSourcesConfiguration", must be one of: type, knowledgeBaseConfiguration
```

I guess I could try/catch for that.

Reply:
Simpler code, but less explicit maybe. Plus it'd help to log the version of boto that folks have so that they know where they're at "<version X> required and found <version Y>". So maybe your approach is better that way.

Reply:
I have added it in, but I did not realize that the "external sources" feature was only recently added in botocore.  I tested it by launching breeze and running `pip install boto3==1.34.90` then running the dag.  So it does work IF the user installs a version of boto which might cause issues with aiobotocore.

The latest version of aiobotocore constrains botocore<1.34.70,>=1.34.41.  

I'm not entirely sure what the best way to handle this would be.  Very open to suggestions and advice on this.